cli/src/test/java/com/box/l10n/mojito/cli/command/PullCommandTest.java

```diff
@@ -896,4 +934,269 @@ public void pullFullyTranslated() throws Exception {
+        checkJapaneseTranslationsInDelta(deltaForBaselineToFrenchAndJapaneseImport);
+
+        logger.debug("Check that the delta for the pull run generated after the French import only return Japanese translations");
+        DeltaResponseDTO deltaForAfterFrenchImportToFrenchAndJapaneseImport = deltaService.getDeltasForRuns(
+                repository,
+                null,
+                ImmutableList.of(pushRun),
+                ImmutableList.of(pullRunAfterFrenchImport));
+
+        Assertions.assertThat(deltaForAfterFrenchImportToFrenchAndJapaneseImport.getTranslationsPerLocale()).hasSize(1);
+        checkJapaneseTranslationsInDelta(deltaForAfterFrenchImportToFrenchAndJapaneseImport);
+
+        logger.debug("Record a pull run after the French and Japanese imports for later delta testing");
+        getL10nJCommander().run("pull", "-r", repository.getName(),
+                "-s", getInputResourcesTestDir("source").getAbsolutePath(),
+                "-t", getTargetTestDir("target_after_french_and_japanese_imports").getAbsolutePath(),
+                "--record-pull-run",
+                "--export-pull-run-id-to-file");
+
+        logger.debug("Simulate commit and linked to pull-run");
+        String pullRunHash3 = "ddaa33";
+        commitService.saveCommit(repository, pullRunHash3, "coder@mail.com", "coder", DateTime.now());
+        getL10nJCommander().run("commit-to-pull-run", "-r", repository.getName(),
+                "-i", getTargetTestDir("target_after_french_and_japanese_imports").getAbsolutePath(),
+                "--commit-hash", pullRunHash3);
+
+        PullRun pullRunAfterFrenchAndJapaneseImports = commitService.getLastPullRun(ImmutableList.of(pullRunHash3), repository.getId())
+                .orElseThrow(() -> new RuntimeException("There must be a pull run"));
+                
+        logger.debug("Update one translation in French and remove one in Japanese");
+        Locale frFR = localeService.findByBcp47Tag("fr-FR");
+        Locale jaJP = localeService.findByBcp47Tag("ja-JP");
+        TMTextUnit tmTextUnit = tmTextUnitRepository.findFirstByAssetIdAndName(asset.getId(), "1_month_duration");
+        tmService.addCurrentTMTextUnitVariant(tmTextUnit.getId(), frFR.getId(), "1 mois -- update");
+        tmTextUnitCurrentVariantService.removeCurrentVariant(tmTextUnitCurrentVariantRepository.findByLocale_IdAndTmTextUnit_Id(jaJP.getId(), tmTextUnit.getId()).getId());
+
+        logger.debug("Check that the delta for the pull run generated returns 1 modifications");
+        DeltaResponseDTO deltaForAfterFrenchAndJapaneseImportsToSmallChanges = deltaService.getDeltasForRuns(
+                repository,
+                null,
+                ImmutableList.of(pushRun),
+                ImmutableList.of(pullRunAfterFrenchAndJapaneseImports));
+
+        // TODO(jean) We may have agreed on returning nothing for delete and have to insert either a fix or the
```

Comment:
Say we return something - would we want to include that in the final output for a downstream client to use? And if so - would we load that up and expect that translation to be displayed to the end user?
I can't think of a scenario atm where that is desirable - and am thinking for that reason to not include any unusable data essentially.

Reply:
the scenario would be an invalid translation that you want to remove. the workaround would be to add the english (or parent) as current translation while re-translation happens but that might be confusing too